### PR TITLE
Update the generic rule for packages to use namespaces

### LINKED ISSUE #176: Bad notification
Hi,

I just received tons of notifications:

* ralph requested package php on branch master
* ralph requested package php on branch f26
* ralph changed ralph's 'watchcommits' permission on php (master) to 'Approved'
* ..

While "php" already exists.

LInk in notification point to https://admin.fedoraproject.org/pkgdb/package/php/ (so obviously wrong)

I think this is related to modularization.

Please fix this

P.S. I also received notification from git-dist with same issue (don't know if this have be fixed)

### PR DESCRIPTION
**Pull request by jeremycline**
This depends on a change in the Fedora Processor changing to accept a
new kwarg, ``namespace``, that accounts for the namespace a package is
in when performing ``msg2packages``.

fixes #176

Note: The tests currently don't pass (since there needs to be a fix in https://github.com/fedora-infra/fedmsg_meta_fedora_infrastructure) and this shouldn't be merged yet.

### CODE REVIEW

File: fmn/rules/generic.py [hunk 1]
```diff
@@ -143,7 +151,11 @@ def user_package_watch_filter(config, message, fasnick=None, *args, **kw):
             return False
         usr_packages = fmn.rules.utils.get_packages_of_user(
             config, fasnick, only_watchcommits)
-        return usr_packages.intersection(msg_packages)
+        for namespace, packages in usr_packages.items():
+            namespaced_msg_packages = fmn.rules.utils.msg2packages(
+                message, namespace=namespace, **config)
+            if packages.intersection(namespaced_msg_packages):
+                return True
```
Comment (bowlofeggs) [inline fmn/rules/generic.py]: This code block seem duplicated with the other two changes in this file. Perhaps make a shared function?

File: fmn/tests/rules/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                'erlang-luerl',
+                'erlang-oauth2',
+                'erlang-p1_iconv',
+                'erlang-p1_mysql',
+                'erlang-p1_oauth2',
+                'erlang-p1_pam',
+                'erlang-p1_pgsql',
+                'erlang-p1_utils',
+                'erlang-p1_xmlrpc',
+                'erlang-p1_zlib',
```
Comment (bowlofeggs) [inline fmn/tests/rules/test_utils.py:77]: Oh yeah, some erlang!

**Review by bowlofeggs — APPROVED** [review]
Just one suggestion.

**Comment by jeremycline** [comment]
@bowlofeggs I've made a few significant changes so it'd be good to have a re-review.

File: fmn/tests/rules/test_generic.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Suite 500, Boston, MA 02110-1335,  USA
+"""Tests for the :mod:`fmn.rules.generic` module."""
+from __future__ import unicode_literals
+
+from fedmsg.meta import make_processors
+import unittest
```
Comment (bowlofeggs) [inline fmn/tests/rules/test_generic.py]: For PEP-8 you might make these two imports be in different groups.

**Review by bowlofeggs — APPROVED** [review]

**Pull request merged**